mailbox/memory/src/main/java/org/apache/james/mailbox/inmemory/mail/InMemoryAnnotationMapper.java

```diff
@@ -91,21 +93,63 @@ public boolean apply(MailboxAnnotation input) {
     }
 
     @Override
+    public List<MailboxAnnotation> getAnnotationsByKeysWithAllDepth(final Set<MailboxAnnotationKey> keys) {
+        return ImmutableList.copyOf(Iterables.filter(retrieveAllAnnotations(mailboxId), getPredicateFilterByAll(keys)));
+    }
+
+    @Override
+    public List<MailboxAnnotation> getAnnotationsByKeysWithOneDepth(final Set<MailboxAnnotationKey> keys) {
+        return ImmutableList.copyOf(Iterables.filter(getAnnotationsByKeysWithAllDepth(keys), getPredicateFilterByOne(keys)));
+    }
+
+    private Predicate<MailboxAnnotation> getPredicateFilterByAll(final Set<MailboxAnnotationKey> keys) {
+        return new Predicate<MailboxAnnotation>() {
+            public boolean apply(final MailboxAnnotation input) {
```

Comment:
Basically it's generated by default but as review comment before, Benoit said that we don't need it. So do you know what is the rule in our team?